crates/e2e/tests/e2e/helpers/services.rs

```diff
@@ -0,0 +1,244 @@
+use {
+    crate::{
+        helpers::{wait_for_condition, Contracts},
+        local_node::NODE_HOST,
+    },
+    clap::Parser,
+    ethcontract::H256,
+    model::{
+        auction::AuctionWithId,
+        order::{Order, OrderCreation, OrderUid},
+        quote::{OrderQuoteRequest, OrderQuoteResponse},
+    },
+    reqwest::{Client, StatusCode},
+    sqlx::Connection,
+    std::time::Duration,
+};
+
+pub const API_HOST: &str = "http://127.0.0.1:8080";
+pub const ORDERS_ENDPOINT: &str = "/api/v1/orders";
+pub const QUOTING_ENDPOINT: &str = "/api/v1/quote";
+pub const ACCOUNT_ENDPOINT: &str = "/api/v1/account";
+pub const AUCTION_ENDPOINT: &str = "/api/v1/auction";
+pub const TRADES_ENDPOINT: &str = "/api/v1/trades";
+pub const VERSION_ENDPOINT: &str = "/api/v1/version";
+
+/// Wrapper over offchain services.
+/// Exposes various utility methods for tests.
+pub struct Services<'a> {
+    contracts: &'a Contracts,
+    http: Client,
+}
+
+impl<'a> Services<'a> {
+    /// Create a new instance of this struct and clear the database.
+    pub async fn new(contracts: &'a Contracts) -> Services<'a> {
+        Self::clear_database().await;
```

Comment:
Not sure if clearing a DB should be part of this constructor. I get that it should happen for every test and this would be an easy way to that under the hood but these 2 things seem unrelated.
I think it would be different if the `Services` struct would create a fresh in-memory DB instance and made every component use that instead. Not sure how easy that would be, though.

Reply:
Yeah, maybe doing this here is not very obvious. Maybe we could move clearing the database to the test wrapper function? That function currently locks and cleans up the hardhat node.
Creating a new DB instance for each test is a bit problematic - we can either use the existing database as a template and clone it inside Postgres (this fails if there are any existing connections to the template database) or create a new database and run migrations (not sure how our migrations work, as they use flyway)

Reply:
Done ^^